packages/codemods/src/codemods/v0.39.x/updateRouterParamTypes/updateRouterParamTypes.ts

```diff
@@ -0,0 +1,145 @@
+import type {
+  API, FileInfo, JSXExpressionContainer,
+  ObjectExpression,
+} from 'jscodeshift'
+
+export default function transform(file: FileInfo, api: API) {
+  const newPropertyName = {
+    constraint: 'match',
+    transform: 'parse',
+  }
+
+  const j = api.jscodeshift
+  const ast = j(file.source)
+
+  const renameParamTypeKey = (paramTypeKey: any) => {
+    // paramTypeKey here could be any one of following marked as 👉
+    /*
+    const slug = {
+      👉 constraint: /\w+-\w+/,
+      👉 transform: (param) => param.split('-'),
+    }
+    ...
+        <Router
+          paramTypes={{
+            slug,
+            embeddedProperties: { 👉 constraint: constraint, 👉 transform },
+            embedded: {
+              👉 constraint: /\w+.\w+/,
+              👉 transform: (param) => param.split('.'),
+            },
+          }}
+        >
+        </Router>
+    ...
+    */
+
+    // To force the value to be explicit. {{ transform }} -> {{ parse: transform }}
+    // paramTypeKey.value = j.identifier.from(paramTypeKey.value)
+    if (paramTypeKey.shorthand) {
+      paramTypeKey.shorthand = false
+    }
+
+    paramTypeKey.key.name =
+      newPropertyName[paramTypeKey.key.name as keyof typeof newPropertyName]
+  }
+
+
+  const mapToNewSyntax = (
+    allParamTypeProperties: ObjectExpression['properties']
+  ) => {
+    // allParamTypeProperties here is array of following marked as 👉
+    /*
+        <Router
+          paramTypes={{
+            👉 slug,
+            👉 embeddedProperties: { constraint: constraint, transform },
+            👉 embedded: {
+                constraint: /\w+.\w+/,
+                transform: (param) => param.split('.'),
+            },
+          }}
+        >
+    */
+
+    allParamTypeProperties.forEach((paramTypeProperty) => {
+      // paramTypeProperty.value could be either ObjectExpression, Identifier
+      if (paramTypeProperty.type === 'SpreadProperty' || paramTypeProperty.type === 'SpreadElement' || paramTypeProperty.type === 'ObjectMethod') {
```

Comment:
@callingmedic911 but, once you accept any objectExpression.property ofcourse it could be one of these weird edge cases (which honestly, I don't think we need to handle).

But I think not having "any" makes it super obvious that we're intentionally skipping these cases. 

I've not changed any of your logic otherwise